### PR TITLE
Add back the missing code that handle source=="vals[i]" code.

### LINKED ISSUE #31330: XmlSerializer fails to deserialize IDREFS into string[]
Class to deserialize into:

```C#
[Serializable()]
[System.Xml.Serialization.XmlType("MsgDocumentType", Namespace = "http://example.com")]
[System.Xml.Serialization.XmlRoot("Document", Namespace = "http://example.com")]
public partial class MsgDocumentType
{
    [System.Xml.Serialization.XmlAttribute("id", DataType = "ID")]
    public string Id { get; set; }

    [System.Xml.Serialization.XmlAttribute("refs", DataType = "IDREFS")]
    public string[] Refs { get; set; }
}
```

XML:

```xml
<?xml version="1.0" encoding="utf-8"?>
<Document xmlns="http://example.com" id="ID1" refs="ID1 ID1 ID1"></Document>
```

This fails in a netcoreapp2.1 with
> InvalidOperationException: The ReadElementContentAsString method is not supported on node type Attribute. Line 2, position 47.

It works ok in net45 and also in netcoreapp2.1 with the Microsoft.XmlSerializer.Generator package (pre-generated serializer).

### PR DESCRIPTION
**Pull request by huanwu**
Fix #31330

@mconnew @Lxiamail @yujayee

### CODE REVIEW

File: src/System.Private.Xml/tests/XmlSerializer/XmlSerializerTests.RuntimeOnly.cs [hunk 1]
```diff
@@ -2952,4 +2952,20 @@ public static void VerifyRestrictionElementForTimeSpanTest()
         Assert.True(element.LastAttribute.Value == expectedAttribute, string.Format("{0}Test failed for wrong output from schema: {0}Expected Output: {1}{0}Actual Output: {2}",
                 Environment.NewLine, baseline, actualOutput));
     }
+
+    [Fact]
+    public static void DeserializeIEREFSIntoStringTest()
```
Comment (mganss) [inline src/System.Private.Xml/tests/XmlSerializer/XmlSerializerTests.RuntimeOnly.cs]: Typo: Should be `DeserializeIDREFSIntoStringTest()`

**Comment by huanwu** [comment]
@mconnew Could you do code review?

**Comment by mconnew** [comment]
The current code looks good, but what about codegen and reflection based serializers?

**Comment by karelz** [comment]
@huanwu any update on the PR? There hasn't been any for last 3 weeks ...

**Comment by huanwu** [comment]
@karelz pending for code review. @mconnew

File: src/System.Private.Xml/tests/XmlSerializer/XmlSerializerTests.cs [hunk 2]
```diff
@@ -1691,6 +1691,22 @@ public static void SerializeWithDefaultValueSetToNegativeInfinityTest()
         Assert.True(result);
     }
 
+    [Fact]
+    public static void DeserializeIEREFSIntoStringTest()
```
Comment (mconnew) [inline src/System.Private.Xml/tests/XmlSerializer/XmlSerializerTests.cs]: Can you fix the naming of this function. It should be I**D**REFS

**Comment by mconnew** [comment]
The code needs to be rebased. Once the typo has been fixed, this LGTM.

**Pull request merged**